RFC/src/RFC-0171_MessageSerialisation.md

```diff
@@ -0,0 +1,186 @@
+* Copying transactions or instructions from cold wallets. The ideal format here is a compact but easy-to-copy format.
+* Peer-to-peer text messaging. This is just a special case of what has already been described, with the message
+  structure containing a unicode `message_text` field.
+
+When sending a message from a human to the network, the following happens:
+
+* The message is deserialised into the native structure.
+* The deserialisation acts as an automatic validation step.
+* Additional validation can be performed.
+* The usual machine-to-machine process is followed as described above.
+
+### Binary serialisation formats
+
+The ideal properties for binary serialisation formats are:
+
+* Widely used across multiple platforms and languages, but with excellent Rust support.
+* Compact binary representation
+* Serialisation "Just Works"(TM) with little or no additional coding overhead.
+
+Several candidates fulfil these properties to some degree.
+
+#### [ASN.1](http://www.itu.int/ITU-T/asn1/index.html)
+
+* Pros:
+  * Very mature (was developed in the 80s)
+  * Large number of implementations
+  * Dovetails into ZMQ nicely
+* Cons:
+  * Limited Rust / Serde support
+  * Requires schema (additional coding overhead if no automated tools for this exist)
+
+
+#### [Message Pack](http://msgpack.org/)
+
+* Pros:
+  * Very compact
+  * Fast
+  * Multiple language support
+  * Good Rust / Serde support
+  * Dovetails into ZMQ nicely
+* Cons:
+  * No metadata support
+
+#### [Protobuf](https://code.google.com/p/protobuf/)
+
+Similar to [Message Pack](#message-pack), but also requires schema's to be written and compiled. Serialisation performance and size
+is similar to Message Pack. Can work with ZMQ but is better designed to be used with gRPC.
+
+#### [Cap'n Proto](http://kentonv.github.io/capnproto/)
+
+Similar to [Protobuf](#protobuf), but claims to be much faster. Rust is supported.
+
+#### Hand-rolled serialisation
+
+[Hintjens recommends](http://zguide.zeromq.org/py:chapter7#Serialization-Libraries) using hand-rolled serialization for
+bulk messaging. While Pieter usually offers sage advice, I'm going to argue against using custom serialisers at this
+point in time for the following reasons:
+* We're unlikely to improve hugely over MessagePack
+* Since serde does 95% of our work for us with MessagePack, there's significant development overhead (and new bugs)
+  involved with a hand-rolled solution.
+* We'd have to write de/serialisers for every language that wants Tari bindings; whereas every major language has a
+  MessagePack implementation.
+
+### Serialisation in Tari
+
+Deciding between these protocols is largely a matter of preference, since there isn't that much to choose between them.
+Given that ZMQ is used in other places in the Tari network; MessagePack looks to be a good fit while offering a compact
+data structure and highly performant de/serialisation. In Rust in particular, there's first-class support for Message
```

Comment:
nit
```suggestion
data structure and highly performant de/serialisation. In Rust in particular, there's first-class support for MessagePack
```

And remove `Pack` from next line

Reply:
:+1: